src/sentry/data_export/tasks.py

```diff
@@ -65,26 +71,46 @@ def assemble_download(
         return
 
     try:
+        # ensure that the export limit is set and capped at EXPORTED_ROWS_LIMIT
         if export_limit is None:
             export_limit = EXPORTED_ROWS_LIMIT
         else:
             export_limit = min(export_limit, EXPORTED_ROWS_LIMIT)
 
-        # if there is an export limit, the last batch should only return up to the export limit
-        if export_limit is not None:
-            batch_size = min(batch_size, max(export_limit - offset, 0))
-
         processor = get_processor(data_export, environment_id)
 
         with tempfile.TemporaryFile() as tf:
             writer = csv.DictWriter(tf, processor.header_fields, extrasaction="ignore")
             if first_page:
                 writer.writeheader()
 
-            rows = process_rows(processor, data_export, batch_size, offset)
-            writer.writerows(rows)
+            # the position in the file at the end of the headers
+            starting_pos = tf.tell()
+
+            # the row offset relative to the start of the current task
+            # this offset tells you the number of rows written during this batch
+            batch_offset = 0
+
+            # the absolute row offset from the beginning of the export
+            next_offset = offset + batch_offset
+
+            while True:
+                # the number of rows to export in the next mini-batch
```

Comment:
```suggestion
                # the number of rows to export in the next batch fragment
```

Not sure if this is clearer, but it felt like we are overloading the word batch here.

Reply:
Yeah i was thinking that when i was writing this but couldn't think of a better word, I'm liking the word fragment though. Let me update this with the word fragment.